src/PlusDataCollection/Atracsys/AtracsysTracker.cxx

```diff
@@ -735,12 +772,41 @@ AtracsysTracker::ATRACSYS_RESULT AtracsysTracker::Connect()
     return ERROR_OPTION_NOT_FOUND;
   }
 
+  // Needs to be after the device option enumeration
+  const ftkOptionsInfo* info;
+  if (!this->GetOptionInfo("Calibration processing datetime", info))
+  {
+    LOG_ERROR("Option unknown: \"Calibration processing datetime\"");
+    return ERROR_OPTION_NOT_FOUND;
+  }
+  ftkBuffer buff;
+  ftkGetData(this->Internal->FtkLib, this->Internal->TrackerSN, info->id, &buff);
+  this->Internal->CalibrationDate = std::string(buff.data);
+
+  // Check whether onboard processing is off or on (spryTrack only)
+  if (this->DeviceType == SPRYTRACK_180 || this->DeviceType == SPRYTRACK_300)
+    if (!this->GetOptionInfo("Enable embedded processing", info))
+    {
+      LOG_WARNING(std::string("Embedded processing not part of the option list."));
+      return ERROR_OPTION_NOT_FOUND;
+    }
+    else
+    {
+      int32 val;
+      ftkGetInt32(this->Internal->FtkLib, this->Internal->TrackerSN, info->id, &val, ftkOptionGetter::FTK_VALUE);
+      isOnboardProcessing = (val == 1) ? true : false;
+      LOG_INFO("Embedded processing is initially " << (isOnboardProcessing ? "enabled" : "disabled"));
+    }
+
   return SUCCESS;
 }
 
 //----------------------------------------------------------------------------
 AtracsysTracker::ATRACSYS_RESULT AtracsysTracker::Disconnect()
 {
+  if (this->Internal->isVirtual)
```

Comment:
Can you ensure that all if statements use curly braces.

You can use Astyle to quickly format everything to match Plus (VTK) coding conventions. https://github.com/PlusToolkit/PlusLib/blob/master/.astylerc

Reply:
Done.

Reply:
Thanks!